tests/PHPStan/Rules/Types/data/invalid-union-with-mixed.php

```diff
@@ -0,0 +1,34 @@
+<?php declare(strict_types=1);
+
+namespace InvalidUnionWithMixed;
+
+class Foo
```

Comment:
what about applying the same rules for phpdoc types?

https://phpstan.org/r/9c3d8fed-52b9-4285-b877-c34ec751ae4c

Reply:
We'd have to give it some more thought, thanks to type normalization `mixed|null` is totally fine in PHPDocs, I don't want to mark that as an error.

Reply:
a special rule only for phpdocs invalid types maybe, maybe hidden behind bleeding edge? hmm

Reply:
Invalid types in PHPDocs like `string&int` are already reported. If you think there is some scenario that isn't covered, open a new feature requests, thanks.

Reply:
As mentioned before, we have https://github.com/phpstan/phpstan/issues/6720 already, so it might be a special thing for void mostly :)

Reply:
Void in return types should be transformed into null after a call.

Reply:
would you say only for phpdoc or in general? currently thinking were the best place is to transform types 🤔

Reply:
In MutatingScope::resolveType where MethodCall/StaticCall/FuncCall are handled.

Reply:
Ok thx, I did exactly that already 😊 looked mostly good with a couple of question marks. Will open a PR when ready.